tests/test_asyncio/test_connection.py

```diff
@@ -112,3 +117,79 @@ async def test_connect_timeout_error_without_retry():
         await conn.connect()
     assert conn._connect.call_count == 1
     assert str(e.value) == "Timeout connecting to server"
+
+
+class FakeStream:
```

Comment:
maybe Mock instead of Fake? But.. yes!

Reply:
Sounds fine, I can move these to a more generic place.  A MockSocket and MockStream